pywbem_mock/_cimrepository.py

```diff
@@ -2485,236 +2521,295 @@ def _pull_operations_disabled(self):
     #
     ####################################################################
 
-    def OpenAssociatorInstancePaths(self, **params):
+    def OpenAssociatorInstancePaths(self, InstanceName, AssocClass=None,
+                                    ResultClass=None, Role=None,
+                                    ResultRole=None,
+                                    FilterQueryLanguage=None, FilterQuery=None,
+                                    OperationTimeout=None, ContinueOnError=None,
+                                    MaxObjectCount=None):
         # pylint: disable=invalid-name
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenAssociatorInstancePaths`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
         self.validate_namespace(namespace)
-        self._validate_open_params(**params)
-        params['ObjectName'] = params['InstanceName']
-        del params['InstanceName']
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout)
 
-        result = self.AssociatorNames(**params)
+        instances = self.AssociatorNames(InstanceName, AssocClass=AssocClass,
+                                         ResultClass=ResultClass, Role=Role,
+                                         ResultRole=ResultRole)
 
-        objects = [] if result is None else [x[2] for x in result[0][2]]
-
-        return self._open_response(objects, namespace,
-                                   'PullInstancePaths', **params)
+        return self._open_response(instances, namespace,
+                                   'PullInstancePaths',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
 
     ####################################################################
     #
     #   OpenAssociatorInstances server request
     #
     ####################################################################
 
-    def OpenAssociatorInstances(self, **params):
+    def OpenAssociatorInstances(self, InstanceName, AssocClass=None,
+                                ResultClass=None, Role=None, ResultRole=None,
+                                IncludeClassOrigin=None,
+                                PropertyList=None, FilterQueryLanguage=None,
+                                FilterQuery=None, OperationTimeout=None,
+                                ContinueOnError=None, MaxObjectCount=None):
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         WBEMConnection.OpenAssociatorInstances
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
```

Comment:
See previous comment about `namespace = self.namespace`. Only this time, the namespace is (optionally) passed in InstanceName.